experimental/rule_inference/utils/rule_utils.py

```diff
@@ -1,44 +1,140 @@
 import json
-from typing import Dict, List
+from typing import Dict, List, Set
+import attr
 
-from comby import Comby
 from polyglot_piranha import Filter, OutgoingEdges, Rule, RuleGraph
 
 
+@attr.s(eq=False)
+class RawFilter:
+    id = attr.ib(type=str, default="0")
+    enclosing_node = attr.ib(type=str, default=None)
```

Comment:
what does this do?

Reply:
Ah this is because the actual objects cannot be pickled :| So I had to create these